R/grate.R

```diff
@@ -903,3 +903,87 @@ gcount_column_order <- function(df) {
       'is_allpublic'
     ))
 }
+
+
+#' Enrich report card matriculation percentages with best guesses at 
+#' graduated students
+#' 
+#' @param df data frame of including subgroup percentages
+#' 
+#' @return data_frame
+#' @export
+enrich_grad_count <- function(df) {
+  
+  if (min(df$end_year) < 2013) stop("end_year needs to be > 2012")
+  
+  grad_count_yrs <- df %>%
+    pull(end_year) %>%
+    unique()
+  
+  # need the year before the earliest year if anything earlier than
+  # 2017 is included in order to get 16 mo grad count
+  if (min(grad_count_yrs) < 2017) {
+    grad_count_yrs <- c(min(grad_count_yrs) - 1, grad_count_yrs)
+  }
+  # but there is no grad count data before 2011
+  grad_count_yrs <- grad_count_yrs[grad_count_yrs > 2011]
+  
+  grad_counts <- map(
+    grad_count_yrs, 
+    function(.x) fetch_grad_count(.x)
+  )
+  
+  grad_counts <- grad_counts %>%
+    bind_rows() %>%
+    select(-c(county_name, district_name, school_name,
+              cohort_count))
+    
+  postsec_rates <- data.frame(
+    end_year = 2012:2019,
+    is_16mo = c(T, T, T, T, T, F, F, F)
+    # data is from National Student Clearinghouse
+  )
+  
+  out <- df %>%
+    # 16 mo matriculation rate requires 2011 grad count which we don't have
+    filter(end_year != 2012) %>%
+    left_join(postsec_rates, by = "end_year") %>%
+    mutate(gc_year = if_else(is_16mo, end_year - 1, as.numeric(end_year))) %>%
+    mutate(
+      subgroup = tolower(subgroup),
+      subgroup = case_when(
+        subgroup == "economically disadvantaged students" ~ "economically disadvantaged",
+        subgroup %in% c("english language learners",
+                        "english learners") ~ "limited english proficiency",
+        subgroup == "students with disabilities" ~ "students with disability",
+        subgroup == "two or more races" ~ "multiracial",
+        subgroup == "native hawaiian" ~ "pacific islander", # or native amer.?
```

Comment:
these are the special population subgroup names, correct?  I like consistency across these PR files - that's good.  Maybe we should make a note about weirdness in how subgroups have been collapsed by DOE across various years and we could print it to the console as a `warning()`?  if that sounds good let's open an issue for that as an enhancement (🚢  🚢  🚢 )